test/new-e2e/tests/containers/main_test.go

```diff
@@ -13,14 +13,21 @@ import (
 	"testing"
 
 	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/runner"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/runner/parameters"
 	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/infra"
 )
 
 var keepStacks = flag.Bool("keep-stacks", false, "Do not destroy the Pulumi stacks at the end of the tests")
 
 func TestMain(m *testing.M) {
 	code := m.Run()
-	if runner.GetProfile().AllowDevMode() && *keepStacks {
+	var initOnly bool
+	initOnlyParameter, err := runner.GetProfile().ParamStore().GetBoolWithDefault(parameters.InitOnly, false)
+	if err == nil {
+		initOnly = initOnlyParameter
+	}
```

Comment:
**Suggestion**
Since you are using a method that places the default to `false` and the zero value for the boolean should be `false` as well, you probably can skip the second variable and add a log for the error

```suggestion
	initOnly, err := runner.GetProfile().ParamStore().GetBoolWithDefault(parameters.InitOnly, false)
	if err != nil {
// log the error instead
	}
```